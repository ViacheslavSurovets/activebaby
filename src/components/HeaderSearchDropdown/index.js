export { default as HeaderSearchDropdown } from './HeaderSearchDropdown';
