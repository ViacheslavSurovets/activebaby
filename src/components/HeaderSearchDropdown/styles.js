import styled, { css } from 'styled-components';
import { searchMediaQueryLow } from '@components/SearchDesktop/Search';
import { tablet, mobile } from '@media';

const stylesDropdownSearch = css`
  justify-content: center;
  align-items: center;
  width: 100%;
  height:7rem;
  background: ${ ( { theme } ) => theme.header.dropdownBackground };
  position: absolute;
  right: 0;
  z-index: 2;
`;

export const DropdownSearch = styled.div`
  display: none;
  
  ${ tablet },
  ${ mobile }{
    ${stylesDropdownSearch};
    transform: translateY(74%);
  }
  
  ${ searchMediaQueryLow }{
    ${stylesDropdownSearch};
    transform: translateY(100%);
}
`;

export const DropdownSearchWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  border-bottom: 1px solid ${ ( { theme } ) => theme.header.searchDropdownBorderBottom };
  height: 90%;
  width: 50%;
  ${mobile}{
    width: 75%;
  }
`;

const stylesDropdownInput = css`
display: flex;  
    border: none;
    outline: none;
    padding-left: .5rem;
    width: 90%; 
    font-size: 3rem;
    align-self: center;
    color: ${ ( { theme } ) => theme.header.searchColor };
`;

export const DropdownInput = styled.input`
  display: none; 
  ${ tablet }{
    ${stylesDropdownInput};
  };
  ${ mobile }{
    ${stylesDropdownInput};
  };
  ${ searchMediaQueryLow }{
    ${stylesDropdownInput};
   }
`;
