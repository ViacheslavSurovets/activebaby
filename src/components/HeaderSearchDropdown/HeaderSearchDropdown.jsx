import React from 'react';
import { useTranslation } from 'react-i18next';
import { SearchIcon } from '@components/icons';
import { DropdownSearchWrapper, DropdownInput, DropdownSearch } from './styles';


const HeaderSearchDropdown = () => {
  const { t } = useTranslation ();
  return (
    <DropdownSearch>
      <DropdownSearchWrapper>
        <DropdownInput type="text" placeholder={ `${ t ( 'header.headerSearchPlaceholder' ) }...` } />
        <SearchIcon width='36' height='36' />
      </DropdownSearchWrapper>
    </DropdownSearch>
  );
};

export default HeaderSearchDropdown;
