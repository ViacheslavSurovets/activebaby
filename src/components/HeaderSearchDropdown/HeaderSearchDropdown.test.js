import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import HeaderSearchDropdown from './HeaderSearchDropdown';
import configureMockStore  from 'redux-mock-store';


const mockStore = configureMockStore();
const store = mockStore({});

describe('HeaderSearchDropdown', ()=> {
  it('HeaderSearchDropdown Snapshot', ()=> {
    expect ( shallow (
      <Provider store={store}>
        <HeaderSearchDropdown  />
      </Provider>
    ) ).toMatchSnapshot();
  });
});
