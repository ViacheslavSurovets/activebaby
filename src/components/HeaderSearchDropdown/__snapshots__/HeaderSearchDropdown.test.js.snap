// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderSearchDropdown HeaderSearchDropdown Snapshot 1`] = `ShallowWrapper {}`;
